src/Core/Language/Pack/Loader/RemoteLanguagePackLoader.php

```diff
@@ -34,35 +36,63 @@
 final class RemoteLanguagePackLoader implements LanguagePackLoaderInterface
 {
     /**
-     * The link from which available languages are retrieved.
+     * The languages repository base path
+     *
+     * @TODO : duplicated from {prestashop_repository}/app/config/config.yml;
+     *         used for install context or other cases where no Symfony container is available
      */
-    public const PACK_LINK = 'http://i18n.prestashop-project.org/translations/%ps_version%/available_languages.json';
+    private const LANG_REPOSITORY_BASE_PATH = 'https://i18n.prestashop-project.org';
 
     /**
-     * @var Version
+     * @var Version Prestashop version
      */
     private $version;
 
     /**
-     * @param Version $version
+     * @var string represents the language repository base path
      */
-    public function __construct(Version $version)
+    private $langRepositoryBasePath;
+
+    /**
+     * @param Version $version Prestashop version
+     * @param string $langRepositoryBasePath should become mandatory in future release
+     */
+    public function __construct(Version $version, string $langRepositoryBasePath = self::LANG_REPOSITORY_BASE_PATH)
     {
         $this->version = $version;
+        $this->langRepositoryBasePath = $langRepositoryBasePath;
```

Comment:
Since this is a "RemoteLanguagePackLoader", this should definitely be a domain and not a base path. If this is injected for test mocking purposes, that would be fixable by injecting a circuit breaker.

I'm not fully convinced by the idea of injecting the domain, though. Think about this class's responsibility: if you can choose the domain, why is the path known and fixed by the class? If you can inject it, is it coherent with the chosen class responsibility?

Reply:
> For me it is common sense to make all application external dependencies configurable.

We agree on that, but that's not my point.

My point is that if you read the class's description, it is responsible for "retrieving language pack data from [a] remote host"

Following this principle, your change is mixing its concerns: this class should either have a hard-coded URL, or not be opinionated about the remote path at all (in that case the URL should be injected). Logically, the URLs should be provided to the class, and it should download the packs by itself, instead of providing the URLs for somebody else.

Also, note that this class is now providing public methods that don't exist in its interface.